src/overlay/OverlayManagerImpl.cpp

```diff
@@ -508,9 +515,32 @@ OverlayManagerImpl::tick()
         auto res = mResolvedPeers.get();
         storePeerList(res.known, false, false);
         storePeerList(res.preferred, true, false);
-        mPeerIPTimer.expires_from_now(PEER_IP_RESOLVE_DELAY);
-        mPeerIPTimer.async_wait([this]() { this->triggerPeerResolution(); },
-                                VirtualTimer::onFailureNoop);
+        auto retries = res.retryCount;
+
+        std::chrono::seconds retryDelay;
```

Comment:
Can you just set retryDelay to PEER_IP_RESOLVE_DELAY here? Then you don't need a condition on line 522 and 536. You can have one if statement `if (hasErrors && retries) ....`

Reply:
I ended up reworking this part as I now use member variable anyways